src/app/components/TaskForm.tsx

```diff
@@ -16,29 +16,46 @@ import {
   DialogTrigger,
 } from "@/components/ui/dialog";
 import { PlusCircle } from "lucide-react";
-import { toast, ToastContainer } from "react-toastify";
-import "react-toastify/dist/ReactToastify.css";
+import { toast } from "react-toastify";
 
 const Categories_List = ["choose", "Work", "Personal", "Health", "Other"];
 
 interface TaskFormProps {
   userId: string;
   addTask: (formData: FormData) => Promise<void>;
+  translateText: (arabicText: string) => Promise<string>;
 }
 
-export default function TaskForm({ userId, addTask }: TaskFormProps) {
+export default function TaskForm({ userId, addTask, translateText }: TaskFormProps) {
   const [task, setTask] = useState({
     title: "",
     description: "",
     category: "choose",
     date: "",
   });
 
+  function containsArabic(text: string): boolean {
```

Comment:
**Suggestion: do this work on the server**

* In the endpoint check. if it has arabic/ if it deos translate that
* otherwise leave it alone
More logic on the server the better imho

Reply:
You can also do both requests in parallel on the server using `Promise.all`

Reply:
done